### ISSUE TITLE
`roots` does not return the correct result for polynomials over towers of field extensions

### ISSUE THREAD

**Issue by HechtiDerLachs** [oscar-system/Oscar.jl#3082]
Try the following code:
```julia
P, t = QQ[:t]
kk = splitting_field(t^3 - 1)
kks, s = kk[:s]
kk = splitting_field(s^7 - 13)

h = -1//4*s^8 - 3//2*s^4 - 1//4
kk2 = splitting_field(h)
hh = map_coefficients(kk2, h)
alpha = roots(hh)
iszero(hh(alpha[1]))
```
I get `false` as the last answer. But if I ask for `roots(hh)` I expect elements which are annihilated by the polynomial `hh`. 
If I say `factor(hh)`, I get a correct result, though.

**Comment by HechtiDerLachs** [comment]
@fieker @thofma : Any ideas what's going on here?

**Comment by thofma** [comment]
Given that the `roots` function calls `factor` in this case, it should be easy to fix. I am looking at it.

**Comment by fieker** [comment]
stupidity: we scale by the leading coff of the poly. Why I really do not know
```
function roots(f::PolyRingElem{<: NumFieldElem})
  lf = factor(f)
  @assert degree(unit(lf)) == 0
  #scale = inv(coeff(unit(lf), 0))
  return elem_type(base_ring(f))[-constant_coefficient(x)*inv(leading_coefficient(x)) for x = keys(lf.fac) if degree(x) == 1]
end
```
works

**Comment by fieker** [comment]
Tommy: you or me?

**Comment by thofma** [comment]
I will open a PR.

**Comment by thofma** [comment]
Fixed in https://github.com/thofma/Hecke.jl/pull/1317. But we need https://github.com/oscar-system/Oscar.jl/pull/3071 to make it available in Oscar.

**Comment by HechtiDerLachs** [comment]
Thanks!

**Comment by thofma** [comment]
Fixed in Hecke 0.31.1, which got released a minute ago.
